<?xml version="1.0"?>
<launch>

  <!-- NOTE: This is an example launch file, please do not modify! -->

  <!-- compare error between two poses -->
  <include file="$(find mbot_pose_evaluator)/ros/launch/pose_evaluator.launch">
    <arg name="pose_1" value="/mbot_pose_evaluator/reference_pose" /> <!-- reference pose -->
    <arg name="pose_2" value="/mbot_pose_evaluator/target_pose" /> <!-- target pose -->
  </include>
    
  <!-- POSE1 : publish reference pose in the origin -->
  <node pkg="rostopic" type="rostopic" name="reference_pose_publisher" output="screen"
        args="pub -r 10 /mbot_pose_evaluator/reference_pose geometry_msgs/PoseStamped '{header:  {seq: 0.1, stamp: {secs: 0, nsecs: 0}, frame_id: 'map'}, pose: {position: {x: 0.0, y: 0.0, z: 0.0}, orientation: {x: 0.0, y: 0.0, z: 0.627936913751, w: 0.778264243267}}}'" />
  
  <!-- publishes the reference frame needed by mockup gui -->
  <node pkg="tf" type="static_transform_publisher" name="base_link_broadcaster" args="0 0 0 0 0 0 map base_link 15" />
  
  <!-- POSE2 : publish a pose, generated by a gui interface which can be controller by scrollbars -->
  <node pkg="mir_pregrasp_planning" type="pose_mock_up_gui_node" name="pose_mock_up_gui" output="screen">
    <!-- the cycle rate of the node (in hz) -->
    <param name="loop_rate" type="double" value="10.0"/>
    <param name="reference_frame" type="str" value="base_link"/>
    <remap from="~mock_up_pose" to="/mbot_pose_evaluator/target_pose"/>
  </node>

  <!-- display the error on terminal -->
  <node pkg="rostopic" type="rostopic" name="error_visualizer" output="screen"
        args="echo /pose_error_calculator/pose_error/angular/z" />
  
</launch>
